_rules/status-text-has-aria-live-mogq50.md

```diff
@@ -0,0 +1,525 @@
+
+This rule applies to any [HTML element][] that has a [text node][] as a [descendant][] in the [flat tree][] if all of the following are true:
+
+- **available**: the element is [included in the accessibility tree][]; and
+- **change**: the element or its [descendant][] [text node][] value is an [event originated change in the content][].
+
+## Expectation 1
+
+There exists at least one test target with an implicit or explicit `aria-live` value of "assertive" or "polite" for [event originated changes in the content][event originated change in the content] that communicate the same message.
+
+## Expectation 2
+
+The [HTML element][] with an implicit or explicit `aria-live` value of "assertive" or "polite" is an [ancestor][] in the [flat tree][] of the [HTML element][] or [text node][] that [changes][event originated change in the content].
+
+## Assumptions
+
+The text changes meet the definition of [status message][]. If this is not the case, success criterion [4.1.3 Status Messages][success criterion 4.1.3 status messages] may be satisfied even if this rule failed.
+
+This rule assumes that the [explicit role][] of the elements does not need to be appropriate to satisfy success criterion [4.1.3 Status Messages][success criterion 4.1.3 status messages]. For example, using `role="alert"` on elements with score updates may be of concern for success criterion [1.3.1 Info and Relationships][success criterion 1.3.1 info and relationships] because the `role="status"` appears to be more appropriate in such a context.
+
+This rule assumes that the changes happen within a 1 minute time span after the event firing and therefore the comparison between the page before and after the [event](https://dom.spec.whatwg.org/#concept-event) [firing](https://dom.spec.whatwg.org/#concept-event-fire) can be made at any time after that time span elapses. If there are changes after this time span, they may not be detected as [changes in content][event originated change in the content] and the rule may pass but [success criterion 4.1.3 Status Messages][] is not satisfied. The arbitrary 1 minute time span, selected so that testing this rule would not be impractical, is not included in WCAG.
+
+## Accessibility Support
+
+There exists a known combination of operating system and browsers that do not specify the implicit `aria-live` property.
+
+## Background
+
+- [ARIA19: Using ARIA role=alert or Live Regions to Identify Errors](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA19.html)
+- [ARIA21: Using role=status to present status messages](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA22.html)
+- [ARIA23: Using role=log to identify sequential information updates](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA23.html)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The `p` element that appears after activating submit button with no value for the first name field has `role="alert"`. `role="alert"` has an implicit `aria-live` value of "assertive".
```

Comment:
I'm not sure test cases like this can work. Up until now, we've only ever used test cases that can be used in their initial loaded stage. If we want tests like this, we're going to need to figure out some way to progamatically describe how to get the page into the state it needs to be tested in.

We should probably have a proposal on how to do this, before we can move forward with this. One way would be to use a script go get the page into the correct state and fire a custom event that implementors can listen for. That has some limitations, since it can't use state that an not be set with JS, such as actually setting `:hover` and `:focus` states. Another solution could be to include some Gherkin with the text case that tools will need to implement and run.